packages/common/src/store/withdraw-usdc/slice.ts

```diff
@@ -0,0 +1,83 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit'
+
+import { Status } from '../../models/Status'
+
+type WithdrawUSDCState = {
+  withdrawStatus: Status
+  destinationAddress?: string
+  amount?: number
+  destinationError?: Error
+  amountError?: Error
+}
+
+const initialState: WithdrawUSDCState = {
+  withdrawStatus: Status.IDLE
+}
+
+const slice = createSlice({
+  name: 'withdraw-usdc',
+  initialState,
+  reducers: {
+    setDestinationAddress: (
```

Comment:
FWIW, I don't think we _need_ action/reducer/sagas for setting destination/amount. It might be nice to keep the component responsible for the validations (or use hooks), and send an action to the store only once they're valid and when we need to start the complex transaction logic.

So much of the "saga-first" approach is for the shared code win with mobile and some useful debugging things, but I don't think we're going to have this on mobile anyway so it _might_ be nice to forgo the saga indirection and keep things more closely colocated.

It would also help prevent stale data in the redux store and removes the need for a special "cleanup" action, as there could be a single action payload that's all used in one saga vs properties set on the redux store state piecemeal that then are used in the saga.

Just something to consider, moreso for future - don't have to change this now!

Reply:
nice good thinking. i was kinda just following patterns but yeah you're probs right.